lib/DDGC/Web/Controller/Admin/Stats.pm

```diff
@@ -246,5 +247,45 @@ sub coupons :Chained('base') :Args(0) {
     ") or die $dbh->errstr;
 }
 
+sub github :Chained('base') {
+    my ($self, $c, $since) = @_;
+
+    #die unless 
+    #    $since eq 'last_week'
+    # || $since eq 'last_month'
+    # || $since eq 'last_90_days';
+
+    $c->add_bc('GitHub');
+
+    my %subtract;
+    %subtract = (hours  => 24);
+    %subtract = (weeks  =>  1) if $since eq 'last_week';
+    %subtract = (weeks  =>  2) if $since eq 'week_before';
+    %subtract = (weeks  =>  3) if $since eq 'week_before_that';
+    %subtract = (months =>  1) if $since eq 'last_month';
+    %subtract = (days   => 90) if $since eq 'last_90_days';
+
+    my $start_date = DateTime->now->subtract(%subtract);
+    my $end_date   = DateTime->now;
+    $end_date = DateTime->now->subtract(weeks => 1) if $since eq 'week_before';
+    $end_date = DateTime->now->subtract(weeks => 2) if $since eq 'week_before_that';
+
+    my @stats = DDGC::Stats::GitHub->report(
+        db      => $c->ddgc->db, 
+        between => [$start_date, $end_date],
+    );
+
+    $c->stash->{stats} = \@stats;
+    $c->stash->{tabs}  = {
+        last_24_hours    => "",
+        last_week        => "",
+        week_before      => "",
+        week_before_that => "",
+        last_month       => "",
+        last_90_days     => "",
+    };
+    $c->stash->{tabs}->{$since} = "selected";
```

Comment:
@kablamo This page should be only visible to admins, right? I don't see any check for that if that's the case. Maybe redirect to /ia if user isn't logged in or is not an admin?